packages/forgetti/src/core/optimizer.ts

```diff
@@ -73,6 +73,7 @@ export default class Optimizer {
   createMemo(
     current: t.Expression,
     dependencies?: t.Expression | (t.Expression | undefined)[] | boolean,
+    isUseRef = false,
```

Comment:
maybe instead of a boolean, you can just pass the union type? You had multiple ternaries after this.